desktop-widgets/tab-widgets/TabDiveEquipment.cpp

```diff
@@ -114,12 +114,11 @@ TabDiveEquipment::~TabDiveEquipment()
 // Refresh the corresponding UI field.
 void TabDiveEquipment::divesChanged(const QVector<dive *> &dives, DiveField field)
 {
-	// If the current dive is not in list of changed dives, do nothing
-	if (!current_dive || !dives.contains(current_dive))
+	if (!parent.currentDiveAffected(dives))
 		return;
 
 	if (field.suit)
-		ui.suit->setText(QString(current_dive->suit));
+		ui.suit->setText(QString(parent.currentDive->suit));
```

Comment:
at first I wanted to criticize this model, but the more I look at it and the resulting code, the more I like it. This is clear and easy and avoids inconsistency pitfalls.